core/libraries/rest_api/controllers/model/Read.php

```diff
@@ -688,19 +688,29 @@ protected function createBareEntityFromWpdbResults(EEM_Base $model, $db_row)
                 );
             } elseif ($field_obj instanceof \EE_Datetime_Field) {
                 $field_value = $field_obj->prepare_for_set_from_db($field_value);
-                $timezone = $field_value->getTimezone();
-                EEH_DTT_Helper::setTimezone($field_value, new DateTimeZone('UTC'));
-                $result[ $field_name . '_gmt' ] = ModelDataTranslator::prepareFieldValuesForJson(
-                    $field_obj,
-                    $field_value,
-                    $this->getModelVersionInfo()->requestedVersion()
-                );
-                EEH_DTT_Helper::setTimezone($field_value, $timezone);
-                $result[ $field_name ] = ModelDataTranslator::prepareFieldValuesForJson(
-                    $field_obj,
-                    $field_value,
-                    $this->getModelVersionInfo()->requestedVersion()
-                );
+                if (is_null($field_value)) {
+                    $gmt_date = $local_date = ModelDataTranslator::prepareFieldValuesForJson(
+                        $field_obj,
+                        $field_value,
+                        $this->getModelVersionInfo()->requestedVersion()
+                    );
```

Comment:
So to be clear, if the field_value is null, then this will use the current_time for both `local_date` and `gmt_date`?  If so, why do we want that?

Reply:
Hmmm, it depends on whether the field allows `null`s or not. If the field does allow `null`, both `$gmt_date` and `$local_date` will be `null`, which I think makes sense.
But if the field doesn't allow `null`s, I think you're right: they'll both end up being the current time. But that would be an unusual circumstance: the field says it doesn't allow null (and that should mean the database column doesn't allow it either) and yet somehow we've got a null here anyway... in that case, I'd be inclined to have an exception, because I can't think of a case where the we're preparing a non-nullable datetime field whose value from the DB is null. So I'm inclined to have an exception if that happens...

Reply:
ok, actually it wasn't too hard to handle, so no exception.
So I can't think of a time when the model field won't allow nulls, but the value from the DB is null anyway, but even if that happens, let's set the value to be the field's default. And even then, the GMT field should be the default time in GMT, and the non-GMT field should be the default time in the site's timezone.